internal/updater/utils/html/parsing.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/andybalholm/cascadia"
+	"golang.org/x/net/html"
+)
+
+var (
+	ErrAttrNotFound = errors.New("matching attribute not found")
+)
+
+func GetText(n *html.Node) string {
+	return n.FirstChild.Data
+}
+
+func GetAttr(n *html.Node, key string) (string, error) {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val, nil
+		}
+	}
+
+	return "", ErrAttrNotFound
+}
+
+func CheckID(n *html.Node, id string) bool {
+	if n.Type == html.ElementNode {
+		s, err := GetAttr(n, "id")
+
+		if err == nil && s == id {
+			return true
+		}
+	}
+
+	return false
+}
+
+func Traverse(n *html.Node, id string) *html.Node {
+	if CheckID(n, id) {
+		return n
+	}
+
+	for c := n.FirstChild; c != nil; c = c.NextSibling {
+		res := Traverse(c, id)
+
+		if res != nil {
+			return res
+		}
+	}
+
+	return nil
+}
+
+func GetNodeByID(n *html.Node, id string) *html.Node {
+	return Traverse(n, id)
+}
+
+func GetNodesBySelector(n *html.Node, cssSelector string) []*html.Node {
+	cascSelector := cascadia.MustCompile(cssSelector)
```

Comment:
Perhaps inject the compiled cascadia thingy in this function?

That way you can do like with `regexp.MustCompile` and have globally defined compiled css selectors, especially since each are constant strings.

Reply:
I'd like to keep the utility function accepting strings if possible - it's much easier to maintain and use.
I have moved the repeated compile/matcher building into its own function `buildMatcherFromSelector`